src/main/java/com/commercetools/sync/inventories/InventorySync.java

```diff
@@ -140,283 +100,202 @@ public InventorySync(@Nonnull final InventorySyncOptions syncOptions) {
-                InventorySyncUtils.buildActions(entry, fixedDraft.get(), syncOptions, typeService);
-            if (!updateActions.isEmpty()) {
-                return inventoryService.updateInventoryEntry(entry, updateActions)
-                    .thenAccept(updatedEntry -> statistics.incrementUpdated())
-                    .exceptionally(exception -> {
-                        statistics.incrementFailed();
-                        syncOptions.applyErrorCallback(format(CTP_INVENTORY_ENTRY_UPDATE_FAILED, draft.getSku(),
-                            SkuChannelKeyTuple.of(draft).getKey()), exception);
-                        return null;
-                    });
-            }
-        } else {
-            statistics.incrementFailed();
-        }
-        return CompletableFuture.completedFuture(null);
-    }
-
-    /**
-     * Tries to create Inventory Entry in CTP project, using {@code draft}.
-     * Before calling API, the channel reference from {@code draft} is replaced, so it points to proper channel ID
-     * in target system.
-     * It either creates inventory entry and increments "created" counter in statistics, or increments "failed" counter
-     * and executes error callback function in case of any exception.
-     *
-     * @param draft draft of new inventory entry
-     * @return {@link CompletionStage} instance that indicates method progress
+     * @return a future which contains an empty result after execution of the update.
      */
     @SuppressFBWarnings("NP_NONNULL_PARAM_VIOLATION") // https://github.com/findbugsproject/findbugs/issues/79
-    private CompletionStage<Void> attemptCreate(final InventoryEntryDraft draft) {
-        final Optional<InventoryEntryDraft> fixedDraft = replaceChannelReference(draft);
-        if (fixedDraft.isPresent()) {
-            return inventoryService.createInventoryEntry(fixedDraft.get())
-                .thenAccept(createdEntry -> statistics.incrementCreated())
+    private CompletionStage<Void> buildUpdateActionsAndUpdate(@Nonnull final InventoryEntry entry,
+                                                              @Nonnull final InventoryEntryDraft draft) {
+        final List<UpdateAction<InventoryEntry>> updateActions =
+            InventorySyncUtils.buildActions(entry, draft, syncOptions);
+        if (!updateActions.isEmpty()) {
+            return inventoryService.updateInventoryEntry(entry, updateActions)
+                .thenAccept(updatedEntry -> statistics.incrementUpdated())
                 .exceptionally(exception -> {
-                    statistics.incrementFailed();
-                    syncOptions.applyErrorCallback(format(CTP_INVENTORY_ENTRY_CREATE_FAILED, draft.getSku(),
-                        SkuChannelKeyTuple.of(draft).getKey()), exception);
+                    final String errorMessage = format(CTP_INVENTORY_ENTRY_UPDATE_FAILED, draft.getSku(),
+                        SkuChannelKeyTuple.of(draft).getKey());
```

Comment:
True, I have refactored the `SkuChannelKeyTuple` implementation to just `InventoryEntryIdentifier` since the only thing we need from it is to have a container for the sku and the supply channel id which both uniqley identify and inventory. Implemented here with tests: 24ee590